src/crimson/osd/osd_operations/internal_client_request.cc

```diff
@@ -0,0 +1,134 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:nil -*-
+// vim: ts=8 sw=2 smarttab expandtab
+
+#include <seastar/core/future.hh>
+
+#include "crimson/osd/osd_operations/internal_client_request.h"
+
+namespace {
+  seastar::logger& logger() {
+    return crimson::get_logger(ceph_subsys_osd);
+  }
+}
+
+namespace crimson::osd {
+
+InternalClientRequest::InternalClientRequest(Ref<PG> pg)
+  : pg(std::move(pg))
+{
+}
+
+InternalClientRequest::~InternalClientRequest()
+{
+  logger().debug("{}: destroying", *this);
+}
+
+void InternalClientRequest::print(std::ostream &) const
+{
+}
+
+void InternalClientRequest::dump_detail(Formatter *f) const
+{
+}
+
+CommonPGPipeline& InternalClientRequest::pp()
+{
+  return pg->client_request_pg_pipeline;
```

Comment:
what's the next step? i assumed that we would have dedicated `CommonPGPipeline` type for internal client request. but seems `PGPipeline` is still used here. so what the purpose of the commit of "crimson/osd: split ClientRequest::PGPipeline into CommonPGPipeline" ?

Reply:
The main reason was to minimize the dependencies on stage handling -- the `InternalClientRequest` doesn't use the `await_map` and the commit refactored it out to a `ClientRequest`-specific derivate of `CommonPGPipeline`.